src/dynarec/la64/dynarec_la64_helper.h

```diff
@@ -1806,4 +1806,75 @@ uintptr_t dynarec64_DF(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
 
 #define PURGE_YMM()
 
+#define ALIGNED_ATOMICxw ((fixedaddress && !(fixedaddress & (((1 << (2 + rex.w)) - 1)))) || BOX64ENV(dynarec_aligned_atomics))
+#define ALIGNED_ATOMICH  ((fixedaddress && !(fixedaddress & 1)) || BOX64ENV(dynarec_aligned_atomics))
+
+
+// lock op related macros
+/*
+  LOCK_3264_CROSS_8BYTE cross 8byte, lock lower part 8byte.
+  use ll.d lock lower 8byte. ld.d load ed to likely emulate atomic 4byte/8byte operation
+    op = atomic function
+    x1 = original ed
+    x2 = wback ed addr
+    x4 = result
+    x5 = 8 byte lower block for sc
+    x6 = aligned addr
+*/
+#define LOCK_3264_CROSS_8BYTE(op, x1, wback, x4, x5, x6) \
+    MV(x6, wback);                                       \
+    BSTRINS_D(x6, xZR, 2, 0);                            \
```

Comment:
why not using `ANDI(x6, wback, ~0b111);` instead of `MV` + `BSTRINS_D` ?

Reply:
that's an odd choice to have the imm 0-extended while most other stuffs are sign extended!